spec/ics-024-host-requirements/README.md

```diff
@@ -149,6 +150,32 @@ Host state machines MUST provide the ability to introspect this stored recent co
 type getStoredRecentConsensusStateCount = () => uint64
 ```
 
+### Commitment Path Introspection
+
+Host chains MUST provide the ability to inspect their commitment path, with `getCommitmentPrefix`:
+
+```
+type getCommitmentPrefix = () => CommitmentPrefix
+```
+
+If there is no substate used within the host chain, `getCommitmentPrefix` can be defined as identity function.
+
+The result `CommitmentPrefix` is the definition of the key-value store's substate definition. 
+With the `root: CommitmentRoot` and `state: CommitmentState` of the host state machine, the following property MUST be preserved:
+
+```
+if provableStore.get(path) == value {
+  prefixedPath = applyPrefix(getCommitmentPrefix(), path)
+  if value != nil {
+    assert(verifyMembership(root, createMembershipProof(state, prefixedPath, value), prefixedPath, value))
+  } else {
+    assert(verifyNonMembership(root, createNonMembershipProof(state, prefixedPath), prefixedPath))
+  }
+}
+```
+
+For a key-value store, the `getCommitmentPrefix` RECOMMENDED to be constant.
```

Comment:
How could it not be constant - what would that mean? Wouldn't the handshake fail then?

Reply:
For example, the host state machine is running in a consensus machine with prefix of `"ibc/"`. During the lifetime of the state machine, it should not be changed, which means a value set at height `1000` on the (sub)key `"connection/connectionid"` should stay under the full key of `"ibc/connection/connectionid"`. If the `getCommitmentPrefix` is not constant and can be changed, this means the prefix can be modified over time, which modifies the full key to something else; `"ibc-module/connection/connectionid"`.

Reply:
Put `RECOMMENDED` considering the forward compatibility of state migration during the chain upgrade / module version upgrade(which may change the commitment prefix).

Reply:
If it changes the commitment prefixes, previous connections will no longer work.

Reply:
If you're talking about upgrades we should discuss that explicitly.